stl/inc/algorithm

```diff
@@ -5118,6 +5118,155 @@ _FwdIt shift_right(_ExPo&&, _FwdIt _First, _FwdIt _Last, _Iter_diff_t<_FwdIt> _P
     // not parallelized as benchmarks show it isn't worth it
     return _STD shift_right(_First, _Last, _Pos_to_shift);
 }
+
+#if _HAS_CXX23 && defined(__cpp_lib_concepts)
+namespace ranges {
+    template <class _It, class _Se>
+    constexpr _It _Shift_left_unchecked(_It _First, const _Se _Last, _Iter_diff_t<_It> _Pos_to_shift) {
+        if (_Pos_to_shift <= 0) {

```

Comment:
I see that both `shift_left` and `shift_right` are special-casing `_Pos_to_shift <= 0`, but the Standardese says "*Preconditions:* `n >= 0` is `true`." Thus, I believe that only `== 0` should be special-cased, and we should `_STL_ASSERT` that the value is non-negative.